plugins/faustwp/includes/replacement/callbacks.php

```diff
@@ -86,49 +89,67 @@ function image_source_replacement( $content ) {
 		"#src=\"{$frontend_uri}/#",
 		'#src="/#',
 	);
+
 	return preg_replace( $patterns, "src=\"{$site_url}/", $content );
 }
 
 add_filter( 'wp_calculate_image_srcset', __NAMESPACE__ . '\\image_source_srcset_replacement' );
 /**
- * Callback for WordPress 'the_content' filter to replace paths to media.
+ * Callback for WordPress 'wp_calculate_image_srcset' filter to replace paths when generating a srcset
+ *
+ * @link https://developer.wordpress.org/reference/functions/wp_calculate_image_srcset/
  *
  * @param array $sources One or more arrays of source data to include in the 'srcset'.
  *
- * @return string One or more arrays of source data.
+ * @return array One or more arrays of source data.
  */
 function image_source_srcset_replacement( $sources ) {
-	$use_wp_domain_for_media = use_wp_domain_for_media();
-	$frontend_uri            = faustwp_get_setting( 'frontend_uri' );
-	$site_url                = site_url();
-
-	/**
-	 * For urls with no domain or the frontend domain, replace with the WP site_url.
-	 * This was the default replacement pattern until Faust 1.2, at which point this
-	 * was adjusted to correct replacement bugs.
-	 */
-	$patterns = array(
-		"#^{$site_url}/#",
+
+	if ( ! is_array( $sources ) || empty( $sources ) ) {
+		return $sources;
+	}
+
+	$replace_media_urls = ! use_wp_domain_for_media();
+	$wp_site_urls       = faustwp_get_wp_site_urls();
+	if ( empty( $wp_site_urls ) ) {
+		return $sources;
+	}
+
+	$wp_media_urls       = faustwp_get_wp_media_urls( $wp_site_urls );
+	$relative_upload_url = faustwp_get_relative_upload_url( $wp_site_urls );
+	$frontend_uri        = faustwp_get_setting( 'frontend_uri' );
+
+	$wp_media_site_url = $frontend_uri . $relative_upload_url;
+	$patterns          = array(
+		"#^{$frontend_uri}/#",
 		'#^/#',
 	);
 
-	$replacement = $frontend_uri;
+	foreach ( $sources as $width => $source ) {
```

Comment:
That would be great to improve this loop. Currently it has some memory/structure issues.
I believe that would be very helpful to have inline comments here to understand what the functionality is doing.

Reply:
No problem at all. I will add a few comments and review to see if we can remove the loop 👍